_posts/2024-02-19-jco-1.0.md

```diff
@@ -0,0 +1,301 @@
+---
+title: "Announcing Jco 1.0"
+author: "Yosh Wuyts"
+date: "2024-02-22"
+github_name: "yoshuawuyts"
+excerpt_separator: <!--end_excerpt-->
+---
+
+We’re happy to announce the 1.0 release of [Jco][jco]: a native Javascript
+WebAssembly toolchain and runtime built for WebAssembly Components and [WASI
+0.2](https://bytecodealliance.org/articles/WASI-0.2) [^wasi-one-liner]. Jco can
+natively run Wasm Components inside of Node.js, making it easy to take libraries
+written in different programming languages and execute them using the Node.js
+runtime. And by implementing the entirety of the WASI 0.2 API surface, those
+components can access the network, filesystem, and other system APIs available
+in the Node.js runtime.
+
+[^wasi-one-liner]: WebAssembly Components are an extension to WebAssembly to
+create strongly typed interfaces between different programs. WASI 0.2 is a
+re-work of the "WebAssembly System Interfaces" based on WebAssembly Components.
+The previous version of WASI was not based on components, and as a result was
+rather limited in what it could express.
+
+<!--end_excerpt-->
+
+Our goal is for Jco to be a comprehensive tool for all Component-related
+operations for JavaScript. With Jco 1.0 we're stabilizing a Node.js runtime for
+Wasm Components, as well as a toolchain to take Wasm Components written in other
+languages and import them into JavaScript. From here on out we hope to continue
+stabilizing more features for Jco. Some features are already available
+experimentally, including: native support for browser and Deno.js JavaScript
+runtimes. As well as native support for compiling JavaScript code into
+WebAssembly.
+
+## JavaScript Toolchains in the Bytecode Alliance
+
+Jco is the third JS toolchain project which is part of the Bytecode Alliance. To
+help people navigate this, here is a quick overview of the three projects and
+what their purpose is at the time of writing:
+
+- [Javy](https://github.com/bytecodealliance/javy): A JavaScript → WASI
+compiler toolchain built for WASI 0.1 using the [QuickJS JavaScript
+engine](https://bellard.org/quickjs/). This work predates Wasm Components, but
+Component support may be added in the future.
+- [ComponentizeJS](https://github.com/bytecodealliance/ComponentizeJS): A
+JavaScript → WASI compiler toolchain built for WASI 0.2 and Wasm Components.
+This project is newer and not yet considered stable.
```

Comment:
It would still be great to get a clear summary of ComponentizeJS here. "Create components on Spidermonkey from JS" is the story. It supports _any_ world - WASI is just one.

Reply:
What is the difference between "any world" and "Wasm Components"?

Reply:
I think the important clarifications to this are:

1. It is not at all WASI specific - it works for any world.
2. (optionally) It is built on Spidermonkey.